### PR TITLE
[WIP]mq (ticdc): fix cdc oom due to checkpointTs stuck.

### LINKED ISSUE #7593: TiCDC stuck and OOM when replicating 100k tables.
### What did you do?

1. start 4 cdc nodes (16c 64gb)
2. create a changefeed that replicate 100k tables
3. changefeed stuck, TiCDC OOM

### What did you expect to see?

TiCDC works normally.

### What did you see instead?

as title.
[100k_table_issue.zip](https://github.com/pingcap/tiflow/files/9999058/100k.zip)

### Versions of the cluster

Upstream TiDB cluster version (execute `SELECT tidb_version();` in a MySQL client):

```console
(paste TiDB cluster version here)
```

Upstream TiKV version (execute `tikv-server --version`):

```console
(paste TiKV version here)
```

TiCDC version (execute `cdc version`):

```console
v6.1.1
```

### PR DESCRIPTION
**Pull request by asddongmen**
### What problem does this PR solve?

Issue Number: close #7593 

### What is changed and how it works?
**Why OOM?**
If there are a huge amount of tables that are replicating, a lot of resolvedTs will be sent to mqSink by calling its [FlushRowChangedEvents](https://github.com/pingcap/tiflow/blob/v6.1.2/cdc/sink/mq/mq.go#L175) method.
This method is being called concurrently by a bunch of goroutines, and every resolvedTs event is sent into an [unbounded channel](https://github.com/pingcap/tiflow/blob/v6.1.2/cdc/sink/mq/mq.go#L185), but only one goroutine [is consuming](https://github.com/pingcap/tiflow/blob/v6.1.2/cdc/sink/mq/mq.go#L200) all these resolvedTs from the unbounded channel.
Meanwhile, there is also intensive lock contention between [Line 367](https://github.com/pingcap/tiflow/blob/v6.1.2/cdc/sink/mq/mq.go#L367) and [Line 216](https://github.com/pingcap/tiflow/blob/v6.1.2/cdc/sink/mq/mq.go#L216), it makes the consuming of resolvedTs harder. Eventually, the remained resolvedTs events in [unbounded channel](https://github.com/pingcap/tiflow/blob/v6.1.2/cdc/sink/mq/mq.go#L185) consume all memory of the machine and lead to OOM.

**How to fix it?**

Remove unbounded channel, use StatefulTs to maintain unflushedTs and flushedTs for every table, which can reduce memory usage and alleviate goroutine contention.

#### Tests 

 - Unit test
 - Integration test
 - Manual test (add detailed scripts or steps below)
 - No code

### Release note 

```release-note
Please refer to [Release Notes Language Style Guide](https://pingcap.github.io/tidb-dev-guide/contribute-to-tidb/release-notes-style-guide.html) to write a quality release note.

If you don't think this PR needs a release note then fill it with `None`.
```

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -167,7 +167,7 @@ integration_test_mysql:
 integration_test_kafka: check_third_party_binary
 	tests/integration_tests/run.sh kafka "$(CASE)" "$(START_AT)"
 
-fmt: tools/bin/gofumports tools/bin/shfmt generate_mock generate-msgp-code generate-protobuf
+fmt: tools/bin/gofumports tools/bin/shfmt #generate_mock generate-msgp-code generate-protobuf
```
Comment (asddongmen) [inline Makefile:170]: I will restore this change later.

File: cdc/sink/mq/mq.go [hunk 2]
```diff
@@ -55,6 +57,7 @@ type mqSink struct {
 	topicManager         manager.TopicManager
 	flushWorker          *flushWorker
 	tableCheckpointTsMap sync.Map
```
Comment (CharlesCheung96) [inline cdc/sink/mq/mq.go:59]: Better to rename this filed to `tableStatefulRtsMap`

**Review by nongfushanquan — COMMENTED** [review]
Please add some comments to explain why cdc would oom  and how this pr will fix it .

File: cdc/model/owner.go [hunk 3]
```diff
@@ -306,3 +307,32 @@ type ProcInfoSnap struct {
 	CfID      ChangeFeedID `json:"changefeed-id"`
 	CaptureID string       `json:"capture-id"`
 }
+
+type StatefulRts struct {
+	flushed   Ts
```
Comment (CharlesCheung96) [inline cdc/model/owner.go:312]: Use `model.ResolvedTs` here to support spliting big txn.

**Pull request closed without merging**